notify/testing.go

```diff
@@ -91,3 +113,104 @@ func (f *fakeNotifier) SendResolved() bool {
+		Config:       alertmanager.FullValidConfigForTesting,
+		Secrets:      alertmanager.FullValidSecretsForTesting,
+	},
+	"dingding": {NotifierType: "dingding",
+		Config: dinding.FullValidConfigForTesting,
+	},
+	"discord": {NotifierType: "discord",
+		Config: discord.FullValidConfigForTesting,
+	},
+	"email": {NotifierType: "email",
+		Config: email.FullValidConfigForTesting,
+	},
+	"googlechat": {NotifierType: "googlechat",
+		Config: googlechat.FullValidConfigForTesting,
+	},
+	"kafka": {NotifierType: "kafka",
+		Config:  kafka.FullValidConfigForTesting,
+		Secrets: kafka.FullValidSecretsForTesting,
+	},
+	"line": {NotifierType: "line",
+		Config:  line.FullValidConfigForTesting,
+		Secrets: line.FullValidSecretsForTesting,
+	},
+	"opsgenie": {NotifierType: "opsgenie",
+		Config:  opsgenie.FullValidConfigForTesting,
+		Secrets: opsgenie.FullValidSecretsForTesting,
+	},
+	"pagerduty": {NotifierType: "pagerduty",
+		Config:  pagerduty.FullValidConfigForTesting,
+		Secrets: pagerduty.FullValidSecretsForTesting,
+	},
+	"pushover": {NotifierType: "pushover",
+		Config:  pushover.FullValidConfigForTesting,
+		Secrets: pushover.FullValidSecretsForTesting,
+	},
+	"sensugo": {NotifierType: "sensugo",
+		Config:  sensugo.FullValidConfigForTesting,
+		Secrets: sensugo.FullValidSecretsForTesting,
+	},
+	"slack": {NotifierType: "slack",
+		Config:  slack.FullValidConfigForTesting,
+		Secrets: slack.FullValidSecretsForTesting,
+	},
+	"teams": {NotifierType: "teams",
+		Config: teams.FullValidConfigForTesting,
+	},
+	"telegram": {NotifierType: "telegram",
+		Config:  telegram.FullValidConfigForTesting,
+		Secrets: telegram.FullValidSecretsForTesting,
+	},
+	"threema": {NotifierType: "threema",
+		Config:  threema.FullValidConfigForTesting,
+		Secrets: threema.FullValidSecretsForTesting,
+	},
+	"victorops": {NotifierType: "victorops",
+		Config: victorops.FullValidConfigForTesting,
+	},
+	"webhook": {NotifierType: "webhook",
+		Config:  webhook.FullValidConfigForTesting,
+		Secrets: webhook.FullValidSecretsForTesting,
+	},
+	"wecom": {NotifierType: "wecom",
+		Config:  wecom.FullValidConfigForTesting,
+		Secrets: wecom.FullValidSecretsForTesting,
+	},
+	"webex": {NotifierType: "webex",
+		Config:  webex.FullValidConfigForTesting,
+		Secrets: webex.FullValidSecretsForTesting,
+	},
+}
+
+type NotifierConfigTest struct {
+	NotifierType string
+	Config       string
+	Secrets      string
+}
+
+func (n NotifierConfigTest) GetRawNotifierConfig(name string) *GrafanaIntegrationConfig {
+	secrets := make(map[string]string)
+	if n.Secrets != "" {
+		err := json.Unmarshal([]byte(n.Secrets), &secrets)
+		if err != nil {
+			panic(err)
+		}
+		for key, value := range secrets {
+			secrets[key] = base64.StdEncoding.EncodeToString([]byte(value))
+		}
+	}
+	return &GrafanaIntegrationConfig{
+		UID:                   fmt.Sprintf("%s-uid", name),
+		Name:                  name,
+		Type:                  n.NotifierType,
+		DisableResolveMessage: true,
```

Comment:
I think it's better for this to be deterministic, but are we losing any usefulness by always setting `DisableResolveMessage` to `true`?

Reply:
The default value (if it's not specified in the json) is false. However, all integrations settings in this test data are explicitly specified. So, specifying `true` is more in line with other settings.